File: src/JavaScript/Descending-Order.js
/**
 * 
 * @param {Number} n 
 * @returns {Number}
 */
function descendingOrder(n) {
  return parseInt(String(n).split('').sort().reverse().join(''));
}


// 同理有
function squareDigits(n) {
  return parseInt(String(n).split('').map(x => x ** 2).join(''));
}
console.log(squareDigits(9812) === 816414);


module.exports = descendingOrder;

File: src/Julia/Descending-Order.jl
## Descending-Order.jl

using Pipe, Test


"""
使n的各位数字降序排列
"""
function descending_order(n::Int)::Int
    @pipe n |> string |> collect |> sort |> reverse |> join |> parse(Int64, _)
end

@test descending_order(42145) == 54421